presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1052,10 +1051,10 @@ private PhysicalOperation visitScanFilterAndProject(
                     metadata,
                     sqlParser,
                     sourceTypes,
-                    concat(singleton(rewrittenFilter), rewrittenProjections),
+                    concat(singleton(rewrittenFilter.orElse(TRUE_LITERAL)), rewrittenProjections),
```

Comment:
do we need to analyze this `TRUE_LITERAL` in case of empty filter.

It seems to me this line should look like:

```java
concat(rewrittenFilter.map(Collections::singleton).orElse(emptyList()), rewrittenProjections),
```